homeassistant/components/generic/config_flow.py

```diff
@@ -281,9 +288,24 @@ async def async_step_user(
                         # The automatically generated still image that stream generates
                         # is always jpeg
                         user_input[CONF_CONTENT_TYPE] = "image/jpeg"
-
-                    return self.async_create_entry(
-                        title=name, data={}, options=user_input
+                    self.cached_user_input = user_input
+                    self.cached_title = name
+
+                    # Register a temporary view so that we can show a preview
+                    self.preview_iterator += 1
+                    preview_id = f"{self.flow_id}_{self.preview_iterator}"
+                    cam = GenericCamera(self.hass, user_input, preview_id, "preview")
+                    self.temp_view = CameraImagePreView(self.hass, cam, preview_id)
+                    self.hass.http.register_view(self.temp_view)
```

Comment:
Don't do it like this, instead have a single view that you can register once. You can pass the flow ID to the URL, and the view should be able to look up the context of the flow

Reply:
Done c9b74e2